src/txmempool.cpp

```diff
@@ -615,6 +623,23 @@ void CTxMemPool::removeUnchecked(txiter it, MemPoolRemovalReason reason)
     } else
         vTxHashes.clear();
 
+    if (::deterministicMNManager) {
+        removeUncheckedProTx(it->GetTx());
```

Comment:
what if `deterministicMNManager` will be nullptr when tx is added, but won't be nullptr when `tx` is removed -- will it work correctly?

Reply:
That would be highly problematic as that shows us that in the order of (de)initialization has been messed with, which is highly unlikely in `init` and the constructors in `setup_common`, meaning, at worst, it's a faulty test where the values of globals are being manually manipulated.

The refactoring of `CTxMemPool` split up from the rest of `CDeterministicMNManager` deglobalization was done for ease of review, eventually, we use the pointer set with `ConnectManagers` instead of using globals directly ([source](https://github.com/dashpay/dash/pull/5966/commits/e210cb734e16e0ecef9601251fe0e074ed86827e#diff-c065d4cd2398ad0dbcef393c5dfc53f465bf44723348892395fffd2fb3bac522R350-R427)), making any modifications explicit.